modules/virtualisation/podman/podman.nix

```
@@ -0,0 +1,276 @@
+        Extra CNI plugin configurations to add to podman's default network.
+      '';
+    };
+
+    machines = lib.mkOption {
+      type = types.attrsOf (types.submodule machineOpts);
+      description =
+        "Virtual machine descriptions when podman is run in on non-linux systems";
+      default = {
+        podman-machine-default = {
+          active = true;
+          cpus = 2;
+          disk_size = 100;
+          memory = 2048;
+        };
+      };
+      example = {
+        intel-x86 = {
+          cpus = 2;
+          disk_size = 200;
+          memory = 4096;
+          image_path = "fedora-coreos-38.20230806.3.0-qemu.x86_64.qcow2.xz";
+          qemu_binary = "$${pkgs.qemu}/bin/qemu-system-x86_64";
+        };
+      };
+      # One and only one machine may be active at any given time
+      apply = machines:
+        assert ((lib.lists.count (machine: machine.active)
+          (lib.attrsets.attrValues machines)) == 1);
+        machines;
+    };
+
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.finalPackage ]
+        ++ lib.optional cfg.enableDockerAlias dockerAlias;
+
+      virtualisation.containers = {
+        enable = true; # Enable common /etc/containers configuration
+        #        containersConf.settings = lib.optionalAttrs cfg.enableNvidia {
+        #          engine = {
+        #            conmon_env_vars =
+        #              [ "PATH=${lib.makeBinPath [ pkgs.nvidia-podman ]}" ];
+        #            runtimes.nvidia =
+        #              [ "${pkgs.nvidia-podman}/bin/nvidia-container-runtime" ];
+        #          };
+        #        };
+      };
+    }
+
+    (mkIf pkgs.stdenv.hostPlatform.isLinux (mkMerge [
+      {
+        systemd.user = {
+
+          services.podman = {
+            Unit = {
+              Description = "Podman API Service";
+              Requires = "podman.socket";
+              After = "podman.socket";
+              Documentation = "man:podman-system-service(1)";
+              StartLimitIntervalSec = 0;
+            };
+            Service = {
+              Type = "exec";
+              KillMode = "process";
+              Environment = ''LOGGING=" --log-level=info"'';
+              ExecStart =
+                [ "" "${cfg.finalPackage}/bin/podman $LOGGING system service" ];
+            };
+
+            Install = { WantedBy = [ "default.target" ]; };
+          };
+
+          sockets.podman = {
+            Unit = {
+              Description = "Podman API Socket";
+              Documentation = "man:podman-system-service(1)";
+            };
+            Socket = {
+              ListenStream = "%t/podman/podman.sock";
+              SocketMode = 660;
+            };
+            Install.WantedBy = [ "sockets.target" ];
+          };
+
+        };
+      }
+      (mkIf cfg.enableDockerSocket {
+        home.sessionVariables."DOCKER_HOST" =
+          "unix:///run/user/$UID/podman/podman.sock";
+      })
+    ]))
```

```suggestion
      (mkIf cfg.enableDockerSocket {
        home.sessionVariables."DOCKER_HOST" =
          "unix:///$XDG_RUNTIME_DIR/podman/podman.sock";
      })
    ]))
```

https://github.com/nix-community/home-manager/pull/2630#discussion_r810568878